src/schedule/timer_domain.c

```diff
@@ -32,33 +80,95 @@ static inline void timer_report_delay(int id, uint64_t delay)
 				clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1);
 
 	tr_err(&ll_tr, "timer_report_delay(): timer %d delayed by %d uS %d ticks",
+#ifdef __ZEPHYR__
+	/* Zephyr schedules on nearest kernel tick not HW cycle */
+	if (ll_delay_us <= CYC_PER_TICK)
```

Comment:
CYC_PER_TICK (which is probably "cycles per tick" is in microseconds? (I don't know Zephyr, asking for clarification...)

Reply:
Yep, it's a Zephyr macro - HW clock cycles per kernel tick which is set in Zephyr Kconfig.

Reply:
But ll_delay_us says it's in microseconds, not in clock cycles. Why compare us with clock cycles?

Reply:
your right, the warning timeout is passed in in ticks but converted to us for the trace. Fixed.